src/data_source/storage/sql/queries/availability.rs

```diff
@@ -117,6 +143,27 @@ where
         Ok(common)
     }
 
+    async fn get_vid_common_metadata(
+        &mut self,
+        id: BlockId<Types>,
+    ) -> QueryResult<VidCommonMetadata<Types>> {
+        let mut query = QueryBuilder::default();
+        let where_clause = query.header_where_clause(id)?;
+        // ORDER BY h.height ASC ensures that if there are duplicate blocks (this can happen when
+        // selecting by payload ID, as payloads are not unique), we return the first one.
+        let sql = format!(
+            "SELECT {VID_COMMON_METADATA_COLUMNS}
+              FROM header AS h
+              JOIN vid AS v ON h.height = v.height
+              WHERE {where_clause}
+              ORDER BY h.height ASC
```

Comment:
why don't we select the most recent one? i.e order by DESC

Reply:
The idea is that the response shouldn't change over time. So if you once queried and got an old payload, and then a new one comes with the same hash, you query again you get the same, old one again. There are still plenty of ways to get that new payload if you want it, you just have to specify height, or block hash.

Tbh it doesn't really matter that much, but we should be consistent with the existing behavior unless we have a good reason to deviate from it

Reply:
sounds good!